File: packages/rook_sdk_apple_health/lib/src/data/mapper/data_source_mappers.dart
import 'package:rook_sdk_apple_health/src/data/proto/protos.pb.dart';
import 'package:rook_sdk_core/rook_sdk_core.dart';

extension DataSourceMappers on DataSourceProto {
  DataSource toDomain() {
    return DataSource(
      name,
      description,
      imageUrl,
      imageUrl,
      connected,
      authorizationUrlIsNull ? null : authorizationUrl,
    );
  }
}

File: packages/rook_sdk_apple_health/ios/Classes/Mapper/DataSourceMappers.swift
//
//  DataSourceMappers.swift
//  rook_sdk_apple_health
//
//  Created by Daniel Nolasco on 08/05/24.
//

import Foundation
import RookSDK

extension RookDataSource {
    
    func toProto() -> DataSourceProto {
        guard let realAuthorizationURL = authorizationURL else {
            return DataSourceProto.with { it in
                it.name = name
                it.description_p = description
                it.image = imageUrl
                it.connected = connected
                it.authorizationURLIsNull = true
            }
        }
        
        return DataSourceProto.with { it in
            it.name = name
            it.description_p = description
            it.image = imageUrl
            it.connected = connected
            it.authorizationURL = realAuthorizationURL
            it.authorizationURLIsNull = false
        }
    }
}
